internal/catalogmetadata/filter/bundle_predicates.go

```diff
@@ -58,11 +59,30 @@ func WithBundleImage(bundleImage string) Predicate[catalogmetadata.Bundle] {
 	}
 }
 
-func Replaces(bundleName string) Predicate[catalogmetadata.Bundle] {
-	return func(bundle *catalogmetadata.Bundle) bool {
-		for _, ch := range bundle.InChannels {
+func LegacySuccessor(installedBundle *catalogmetadata.Bundle) Predicate[catalogmetadata.Bundle] {
+	isSuccessor := func(candidateBundleEntry declcfg.ChannelEntry) bool {
+		if candidateBundleEntry.Replaces == installedBundle.Name {
+			return true
+		}
+		for _, skip := range candidateBundleEntry.Skips {
+			if skip == installedBundle.Name {
+				return true
+			}
+		}
+		if candidateBundleEntry.SkipRange != "" {
+			installedBundleVersion, _ := installedBundle.Version()
+			skipRange, _ := bsemver.ParseRange(candidateBundleEntry.SkipRange)
```

Comment:
The difference between blang semver and masterminds semver import aliases is so little that overlooking the subtlety is really easy to do.  
Your code is fine, though I can't help but wonder though when someone will 'cross the streams' and we'll miss it.

Reply:
Yeah agreed. Perhaps we should think about adding some wrappers that obviate the subtlety?